crates/rust-analyzer/src/config.rs

```diff
@@ -929,6 +930,18 @@ impl Config {
         self.workspace_roots.extend(paths);
     }
 
+    pub fn add_linked_projects(&mut self, projects: Vec<ProjectJsonData>) {
+        let linked_projects = &mut self.client_config.global.linkedProjects;
+
+        let mut new_projects: Vec<_> =
+            projects.into_iter().map(ManifestOrProjectJson::ProjectJson).collect();
```

Comment:
i don't think that a `mut` can be dropped in the first borrow, but i've deferred collecting until the match arms.

Reply:
Ah sorry, with the borrow I meant instead of
```diff
-Some(projects) => projects.append(&mut new_projects.collect::<Vec<_>>()),
+Some(projects) => projects.extend(new_projects),
```
didn't realize you were using append, was thinking of extend :)